homeassistant/components/http/__init__.py

```diff
@@ -563,17 +587,51 @@ async def start(self) -> None:
         )
         await self.runner.setup()
 
-        self.site = HomeAssistantTCPSite(
-            self.runner, self.server_host, self.server_port, ssl_context=self.context
-        )
+        socket_path: str | None = None
+        if self.server_host and self.server_host[0].startswith("unix:"):
+            socket_path = self.server_host[0].removeprefix("unix:")
+            self.site = web.UnixSite(
+                self.runner,
+                socket_path,
+                ssl_context=self.context,
+            )
+        else:
+            self.site = HomeAssistantTCPSite(
+                self.runner,
+                self.server_host,
+                self.server_port,
+                ssl_context=self.context,
+            )
         try:
             await self.site.start()
         except OSError as error:
             _LOGGER.error(
                 "Failed to create HTTP server at port %d: %s", self.server_port, error
             )
 
-        _LOGGER.info("Now listening on port %d", self.server_port)
+        if socket_path is not None:
+            # They didn't find a way to put this in aiohttp yet so we have to do it here
+            # https://github.com/aio-libs/aiohttp/issues/4155#issuecomment-643509809
+            if self.socket_permissions is not None:
+                try:
+                    os.chmod(socket_path, self.socket_permissions)
+                except OSError as error:
+                    _LOGGER.error(
+                        "Failed to change permissions on %s: %s", socket_path, error
+                    )
+            if self.socket_user is not None or self.socket_group is not None:
+                try:
+                    shutil.chown(
+                        socket_path,
+                        self.socket_user or -1,
+                        self.socket_group or -1,
+                    )
+                except OSError as error:
+                    _LOGGER.error(
+                        "Failed to change user/group on %s: %s", socket_path, error
+                    )
```

Comment:
This does blocking I/O in the event loop

Reply:
Ah yes that's a pretty silly thing to just forget.